File: LabourBudgetCalculator/Models/DailyHourBreakdown.cs
// DailyHourBreakdown.cs
// Purpose: Data structure to hold the categorized daily hours.
// This can be placed in your Models folder or a new Utilities folder.

namespace LabourBudgetCalculator.Models // Or your appropriate namespace
{
    public class DailyHourBreakdown
    {
        public decimal RegularLabourHours { get; set; }
        public decimal OvertimeLabourHours { get; set; }
        public decimal PremiumLabourHours { get; set; }
        public decimal RegularTravelHours { get; set; }
        public decimal OvertimeTravelHours { get; set; }
        public decimal PremiumTravelHours { get; set; }

        public DailyHourBreakdown()
        {
            RegularLabourHours = 0m;
            OvertimeLabourHours = 0m;
            PremiumLabourHours = 0m;
            RegularTravelHours = 0m;
            OvertimeTravelHours = 0m;
            PremiumTravelHours = 0m;
        }

        // New method to get total planned hours
        public decimal GetTotalPlannedHours()
        {
            return RegularLabourHours + OvertimeLabourHours + PremiumLabourHours +
                   RegularTravelHours + OvertimeTravelHours + PremiumTravelHours;
        }
    }
}

File: src/models/DailyHourBreakdown.ts
export interface DailyHourBreakdown {
  regularLabourHours: number;
  overtimeLabourHours: number;
  premiumLabourHours: number;
  regularTravelHours: number;
  overtimeTravelHours: number;
  premiumTravelHours: number;
}

export class DailyHourBreakdownClass implements DailyHourBreakdown {
  regularLabourHours: number = 0;
  overtimeLabourHours: number = 0;
  premiumLabourHours: number = 0;
  regularTravelHours: number = 0;
  overtimeTravelHours: number = 0;
  premiumTravelHours: number = 0;

  constructor(data?: Partial<DailyHourBreakdown>) {
    if (data) {
      Object.assign(this, data);
    }
  }

  // Method to get total planned hours
  getTotalPlannedHours(): number {
    return this.regularLabourHours + this.overtimeLabourHours + this.premiumLabourHours +
           this.regularTravelHours + this.overtimeTravelHours + this.premiumTravelHours;
  }
} 